src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +119,37 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  /*
+   * Implemented by means of a buffer to keep track of the last n elements during iteration.
+   */
+  def takeRight(n: Int): Iterator[A] = {
+    if (n == 0) Iterator.empty
+    else {
+      val buffer = ArrayBuffer[A]()
```

Comment:
It may be beneficial to set an initial size for the buffer, at least when `n` is less than the default buffer size, possibly also when it is within a reasonable size (e.g. default buffer size is 16, AFAIR, but we could pre-allocate for `n` up to 64 or so)